docker/ubuntu-cuda/Dockerfile

```diff
@@ -18,7 +18,7 @@ RUN apt-get update && apt-get install -y \
     libhdf5-openmpi-dev \
     doxygen \
     vim \
-&& pip2 install cython sphinx sphinxcontrib-bibtex numpydoc --upgrade \
+&& pip install --upgrade pip; pip2 install cython sphinx sphinxcontrib-bibtex numpydoc --upgrade \
```

Comment:
Why do you upgrade pip (3), and then use pip2?

Reply:
The default is pip2 here